rust/src/connection/transaction_stream.rs

```diff
@@ -90,10 +87,17 @@ impl TransactionStream {
         self.options
     }
 
-    pub(crate) fn on_close(&self, callback: impl FnOnce(Option<Error>) + Send + Sync + 'static) {
+    pub(crate) fn on_close(
+        &self,
+        callback: impl FnOnce(Option<Error>) + Send + Sync + 'static,
+    ) -> impl Promise<'_, Result<()>> {
         self.transaction_transmitter.on_close(callback)
     }
 
+    pub(crate) fn close(&self) -> impl Promise<'_, Result<()>> {
```

Comment:
So here's a question: why is it ok for close() to be a true promise, while the variants below for commit and rollback are resolved and re-emitted as promises? We should probably pick 1 approach

cc @farost  @dmitrii-ubskii

Reply:
Yeah we do - but should we? I think close() definitely should be blocking unless the user wants it. Rollback I think probably makes the most sense to block - otherwise the transaction might be in a weird state when you keep using it? Although I think we preserve message ordering so it might be fine not to resolve?

Then there's commit - I think resolving the commit probably makes sense, since users could very easily async fire off huge amounts of say write transactions and there's no slowing down the client at all. Too easy to forget the commit().await/resolve